test/e2e/pods.go

```diff
@@ -60,24 +60,25 @@ func runLivenessTest(c *client.Client, ns string, podDescr *api.Pod, expectResta
 	initialRestartCount := api.GetExistingContainerStatus(pod.Status.ContainerStatuses, "liveness").RestartCount
 	By(fmt.Sprintf("Initial restart count of pod %s is %d", podDescr.Name, initialRestartCount))
 
-	// Wait for at most 48 * 5 = 240s = 4 minutes until restartCount is incremented
-	restarts := false
-	for i := 0; i < 48; i++ {
-		// Wait until restartCount is incremented.
-		time.Sleep(5 * time.Second)
+	// Wait for the restart state to be as desired.
+	restarts, deadline := false, time.Now().Add(2*time.Minute)
+	for start := time.Now(); time.Now().Before(deadline); time.Sleep(2 * time.Second) {
```

Comment:
I avoided using `wait.Poll` because it is terrible (#10307) and because it's not used here I thought I could get away with not switching to it :-)